### PR TITLE
Execute parameters

### PR DESCRIPTION
**Pull request by levydsa**
Depends on #5

### CODE REVIEW

File: libsql/src/androidTest/java/tech/turso/libsql/LibsqlTest.java [hunk 1]
```diff
@@ -10,23 +13,35 @@ public class LibsqlTest {
...
     public void queryEmptyParameters() {
         try (var db = Libsql.open(":memory:");
-                var conn = db.connect(); ) {
+                var conn = db.connect() ) {
             conn.query("select 1");
         }
     }
 
+    @Test
+    public void failOpenEmptyName() {
```
Comment (haaawk) [inline libsql/src/androidTest/java/tech/turso/libsql/LibsqlTest.java]: This test does something else than what its name suggests

File: libsql/src/main/kotlin/tech/turso/libsql/Connection.kt [hunk 2]
```diff
@@ -13,7 +13,7 @@ class Connection internal constructor(private var inner: Long) : AutoCloseable {
 
     fun query(
         sql: String,
-        params: Map<String?, Value?>?,
+        params: Map<String, Value>,
```
Comment (haaawk) [inline libsql/src/main/kotlin/tech/turso/libsql/Connection.kt:46]: Value should probably be allowed to be null, no?

File: libsql/src/main/kotlin/tech/turso/libsql/Connection.kt [hunk 3]
```diff
@@ -28,7 +28,7 @@ class Connection internal constructor(private var inner: Long) : AutoCloseable {
 
     fun query(
         sql: String,
-        vararg params: Value?,
+        vararg params: Value,
```
Comment (haaawk) [inline libsql/src/main/kotlin/tech/turso/libsql/Connection.kt:61]: Shouldn't it be allowed to pass a null value as param? Maybe not but I'm not sure if null is actually a problem here

File: libsql/src/main/kotlin/tech/turso/libsql/Connection.kt [hunk 4]
```diff
@@ -3,7 +3,37 @@ package tech.turso.libsql
...
+                .setNamed(NamedParameters.newBuilder().putAllParameters(params))
+                .build()
+                .toByteArray()
+
+        nativeExecute(this.inner, sql, buf)
+    }
+
+    fun execute(
+        sql: String,
+        vararg params: Value,
```
Comment (haaawk) [inline libsql/src/main/kotlin/tech/turso/libsql/Connection.kt:23]: Should we allow nulls here?

File: libsql/src/main/kotlin/tech/turso/libsql/Connection.kt [hunk 5]
```diff
@@ -3,7 +3,37 @@ package tech.turso.libsql
 class Connection internal constructor(private var inner: Long) : AutoCloseable {
     fun execute(sql: String) {
         require(this.inner != 0L) { "Attempted to execute with a closed Connection" }
-        nativeExecute(this.inner, sql)
+        nativeExecute(this.inner, sql, byteArrayOf())
+    }
+
+    fun execute(sql: String, params: Map<String, Value>) {
```
Comment (haaawk) [inline libsql/src/main/kotlin/tech/turso/libsql/Connection.kt:9]: Should we allow null values here?

File: libsql/src/androidTest/java/tech/turso/libsql/LibsqlTest.java (hunk 1 above)
Comment (levydsa) [reply libsql/src/androidTest/java/tech/turso/libsql/LibsqlTest.java]: I think i fixed this in another PR, but I will double check.

**Pull request merged**